src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -317,32 +317,33 @@ trait EvalContextPrivExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         let mut tree_borrows = alloc_extra.borrow_tracker_tb().borrow_mut();
         let prot = new_perm.protector.is_some();
 
-        // Store initialized permissions and their corresponding range.
+        // Store initial permissions and their corresponding range.
         let mut perms_map: RangeMap<LocationState> = RangeMap::new(
-            base_offset + ptr_size,
+            ptr_size,
             LocationState::new_uninit(
                 Permission::new_disabled(),
                 foreign_access_skipping::IdempotentForeignAccess::None,
             ),
         );
+        // Keep track of whether the node has any part that allows for interior mutability.
+        let mut is_frozen = true;
 
-        let NewPermission { freeze_perm, freeze_access, nonfreeze_perm, nonfreeze_access, .. } =
-            new_perm;
         this.visit_freeze_sensitive(place, ptr_size, |mut range, frozen| {
+            is_frozen = frozen;
             // Adjust range.
             range.start += base_offset;
```

Comment:
This is no longer correct now that `perms_map` is relative to the place, I think?

Reply:
It is quite strange that this does not cause any test failures. Can you investigate that?